Controller/Heroes/Baccarat/CharacterCards/AceOfSwordsSwordsBaccaratCharacterCardController.cs

```diff
@@ -0,0 +1,158 @@
+    {
+        public AceOfSwordsBaccaratCharacterCardController(Card card, TurnTakerController turnTakerController) : base(card, turnTakerController)
+        {
+
+        }
+
+        private List<Card> actedHeroes;
+
+        public override IEnumerator UseIncapacitatedAbility(int index)
+        {
+            switch (index)
+            {
+                case 0:
+                    {
+                        //Select a card in a trash. Destroy a card in play with the same name.
+                        SelectCardDecision selectDecision = new SelectCardDecision(GameController, base.HeroTurnTakerController, SelectionType.DestroyCard, base.FindCardsWhere(new LinqCardCriteria((Card c) => c.IsInTrash)), cardSource: base.GetCardSource());
+                        IEnumerator coroutine = base.GameController.SelectCardAndDoAction(selectDecision, (SelectCardDecision decision) => base.GameController.SelectAndDestroyCard(base.HeroTurnTakerController, new LinqCardCriteria((Card c) => decision.SelectedCard.Identifier == c.Identifier && c.IsInPlayAndHasGameText, "selected card"), false, cardSource: base.GetCardSource()));
+                        if (base.UseUnityCoroutines)
+                        {
+                            yield return base.GameController.StartCoroutine(coroutine);
+                        }
+                        else
+                        {
+                            base.GameController.ExhaustCoroutine(coroutine);
+                        }
+                        yield break;
+                    }
+                case 1:
+                    {
+                        //One target deals itself 1 energy damage.
+                        SelectCardDecision selectDecision = new SelectCardDecision(GameController, base.HeroTurnTakerController, SelectionType.DealDamage, base.FindCardsWhere(new LinqCardCriteria((Card c) => c.IsTarget)), cardSource: base.GetCardSource());
+                        IEnumerator coroutine2 = base.GameController.SelectCardAndDoAction(selectDecision, (SelectCardDecision decision) => base.DealDamage(decision.SelectedCard, decision.SelectedCard, 1, DamageType.Energy, cardSource: base.GetCardSource()));
+                        if (base.UseUnityCoroutines)
+                        {
+                            yield return base.GameController.StartCoroutine(coroutine2);
+                        }
+                        else
+                        {
+                            base.GameController.ExhaustCoroutine(coroutine2);
+                        }
+                        break;
+                    }
+                case 2:
+                    {
+                        //Each hero character may deal themselves 1 toxic damage to draw a card now.
+                        this.actedHeroes = new List<Card>();
+                        IEnumerable<Function> functionsBasedOnCard(Card c) => new Function[]
```

Comment:
I'm not sure you need this private variable if you instead use SelectTurnTakersAndDoAction.  That will also give you a place to select which of the turn taker's character cards take the damage if that's required.  There's also YesNoAction_ManyPlayers that looks even better.

Reply:
This is how Scholar's Proverbs and Axioms implements it